ChaosMod/Effects/db/Peds/PedsNailguns.cpp

```diff
@@ -0,0 +1,84 @@
+#include <stdafx.h>
+
+static std::vector<Ped> frozenPeds;
+static std::map<Ped, Entity*> pedGuns;
+static std::map<Ped, Entity*> heldWeapons;
+
+static bool isFrozen(const Ped ped) {
+	return std::find(frozenPeds.begin(), frozenPeds.end(), ped) != frozenPeds.end();
+}
+
+static bool hasNailgun(const Ped ped) {
+	return pedGuns.count(ped) > 0;
+}
+
+static void giveNailgun(const Ped ped) {
+	int bone = GET_PED_BONE_INDEX(ped, 0xDEAD); // (kolyaventuri): Right Hand
+
+	Entity nailgun = CREATE_OBJECT(1854391800, 0, 0, 0, false, false, false); // (kolyaventuri): 1854391800 == nailgun hash
+	ATTACH_ENTITY_TO_ENTITY(nailgun, ped, bone, 0.15f, 0.05f, 0.01f, 70.0f, 0.0f, 180.f, true, true, false, false, 2, true);
```

Comment:
I would like to get this to set the entity rotation to match the vector of the held weapon... but I couldn't get that to work quite yet

Reply:
Maybe SET_ENTITY_ROTATION is what you are looking for? https://runtime.fivem.net/doc/natives/?_0x8524A8B0171D5E07 Get the players rotation when shot and if the bullet hits the ped set there rotation to the players.

Reply:
I played around with that, but to no avail.. might need to tinker with it more. I was thinking more along the lines of, the nailgun is in the correct orientation that it seems as though it could be the actual gun. Regardless of the wielded weapon (handgun, sniper, minigun, etc.)